stan/math/prim/fun/Eigen.hpp

```diff
@@ -24,6 +33,7 @@
 #include <Eigen/QR>
 #include <Eigen/src/Core/NumTraits.h>
 #include <Eigen/SVD>
+#include <unsupported/Eigen/MatrixFunctions>
```

Comment:
IMO we should include these unsupported headers only in the files that need them, like we do for e.g. FFT